recidiviz/ingest/us_in_vigo/us_in_vigo_scraper.py

```
@@ -0,0 +1,83 @@
+# Recidiviz - a platform for tracking granular recidivism metrics in real time
+# Copyright (C) 2019 Recidiviz, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# =============================================================================
+
+"""Scraper implementation for us_in_vigo."""
+from typing import Optional
+
+from recidiviz.common.constants.bond import BondType
+from recidiviz.common.constants.charge import ChargeStatus
+from recidiviz.common.constants.charge import CourtType
+from recidiviz.ingest.models.ingest_info import IngestInfo
+from recidiviz.ingest.task_params import Task
+from recidiviz.ingest.vendors.jailtracker.jailtracker_scraper import\
+    JailTrackerScraper
+
+
+class UsInVigoScraper(JailTrackerScraper):
+    """Scraper implementation for us_in_vigo."""
+    def __init__(self, ):
+        super(UsInVigoScraper, self).__init__('us_in_vigo')
+
+    def get_jailtracker_index(self):
+        return 'Vigo_County_IN'
+
+    # pylint:disable=arguments-differ
+    def populate_data(self, content, task: Task,
+                      existing_ingest_info: IngestInfo) -> Optional[IngestInfo]:
+        ingest_info: Optional[IngestInfo] = super(
+            UsInVigoScraper, self).populate_data(content, task,
+                                                 existing_ingest_info)
+
+        if not ingest_info:
```

Should we log an error, or is this possible/normal?

---

I haven't noticed this pattern in other vendors.

---

Base Scraper will throw an error if this happens I believe.

---

Then why is the return type optional?

---

cc @colincadams - some scrapers return `None` because there wasn't data. I think this is the NY case - we might hit a disambiguation page and need to return more tasks instead of returning an IngestInfo.

BaseScraper throws an error if the returned ScrapedData has no IngestInfo and there are no more tasks.

---

So for this scraper we should probably throw an error.

---

Done!